### DISCUSSION TITLE
Shields.io WPM badges

### DISCUSSION THREAD

**Discussion by dashezup** [monkeytypegame/monkeytype#1530] [Ideas] [answered]
Hi what do you think of providing a public API endpoint for get WPM info for sharing WPM badges? The only service I know for sharing badges is https://shields.io

A sample badge may be like this (this is my coding time statistics from my wakapi instance)

![A sample shields io badge](https://img.shields.io/endpoint?url=https://ezup.dev/wakapi/api/compat/shields/v1/dashezup/interval:today&style=flat-square&color=blue&label=today)

seems it's quite simple, just a url to return JSON data with very a few of fields. disable this public API for users by default and allow users to enable this so that they can share such badges through somewhere like github profile README.

Also, do you have any plan/intention on providing API for developers to make custom applications/bots with it? maybe mostly for viewing statics, maybe it's kind of like wakapi/wakatime for viewing coding statics.

**Comment by Miodec** [comment, marked as answer]
Something like that should be definitely possible once the mongo branch is merged in
